File: proyecto-backend-nexu/src/main/java/com/nexu/Users/dto/UsersResponseDto.java
package com.nexu.Users.dto;

import com.nexu.Users.domain.UserRole;
import com.nexu.Users.domain.UserStatus;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;


@Getter
@Setter
@NoArgsConstructor
public class UsersResponseDto {

    private Long id;

    private String email;

    private String name;

    private UserStatus status;

    private UserRole role;

}

File: proyecto-frontend-web-nexu3/src/interfaces/user/UsersResponseDto.ts
export interface UsersResponseDto {
	id: number;
	email: string;
	name: string;
	status: UserStatus;
	role?: UserRole;
}

export type UserRole = 'USER' | 'ADMIN' | 'VIEWER';
export type UserStatus = 'ACTIVE' | 'INACTIVE' | 'SUSPENDED' | 'DELETE';